fdbserver/ClusterController.actor.cpp

```diff
@@ -696,8 +696,8 @@ class ClusterControllerData {
 		auto proxies = getWorkersForRoleInDatacenter( dcId, ProcessClass::Proxy, req.configuration.getDesiredProxies()-1, req.configuration, id_used, first_proxy );
 		auto resolvers = getWorkersForRoleInDatacenter( dcId, ProcessClass::Resolver, req.configuration.getDesiredResolvers()-1, req.configuration, id_used, first_resolver );
 
-		proxies.push_back(first_proxy.worker);
```

Comment:
As an alternative implementation, could you have `getWorkersForRoleInDatacenter` insert this into the results before any others? It seems like we're passing it in and then otherwise excluding it from the results.

Reply:
good idea, I like it